routers/api/v1/repo/pull.go

```diff
@@ -805,6 +807,22 @@ func MergePullRequest(ctx *context.APIContext) {
 		return
 	}
 
+	if form.MergeWhenChecksSucceed {
+		scheduled, err := automerge.ScheduleAutoMerge(ctx, ctx.Doer, pr, repo_model.MergeStyle(form.Do), form.MergeTitleField)
+		if err != nil {
+			if pull_model.IsErrAlreadyScheduledToAutoMerge(err) {
+				ctx.Error(http.StatusConflict, "ScheduleAutoMerge", err)
+				return
+			}
+			ctx.Error(http.StatusInternalServerError, "ScheduleAutoMerge", err)
```

Comment:
```suggestion
			status := http.StatusInternalServerError
			if pull_model.IsErrAlreadyScheduledToAutoMerge(err) {
				status = http.StatusConflict
			}
			ctx.Error(status, "ScheduleAutoMerge", err)
```

Reply:
TBH I think it's not worth to do this trick, indeed it doesn't save any code. 

The old way is more friendly to be searched & grep-ed.

Reply:
How exactly would you grep for status codes?
Would you grep for `ctx.Error\($STATUS`, or would you simply grep for `$STATUS`?
Because for me, it's probably `$STATUS`, which is still possible with this implementation...
If it were the first one, I'd agree, it would be more greppable, but only if you assume that nowhere in the code another format could be used.

Reply:
that's why I wrote it that way :)